robotoff/spellcheck/patterns/__init__.py

```diff
@@ -0,0 +1,38 @@
+from typing import Dict
+from pathlib import Path
+
+from robotoff.spellcheck.base_spellchecker import BaseSpellchecker
+
+Patterns = Dict[str, str]
+
+
+class PatternsSpellchecker(BaseSpellchecker):
+
+    PATTERNS_PATHS = {"fr": Path(__file__).parent / "patterns_fr.txt"}
+
+    def __init__(self, lang: str = "fr"):
+        self.lang = lang
+        self.patterns: Patterns = {}
+
+        with self.PATTERNS_PATHS[lang].open() as f:
+            current_pattern = None
+            for line in f.readlines():
+                line = line.strip().split("#")[0]
+                if len(line) == 0:
+                    current_pattern = None
+                elif current_pattern is None:
+                    current_pattern = line
+                else:
+                    self.patterns[line.lower()] = current_pattern.lower()
```

Comment:
I spotted a bug due to the capitalization:
```
In [4]: len("ﬁ".upper())                                                                                                              
Out[4]: 2

In [5]: "ﬁ".upper() == "FI"                                                                                                           
Out[5]: True
```
Which lead to this incorrect correction: 
```
     "text": "2 FILETS DE POULET"
     "corrected": "2 FiLETS DE POULET"
```

Reply:
Very sneaky error. I changed the way we are processing the case to avoid such an error.
I also added some tests for this specific function.
Commit : https://github.com/openfoodfacts/robotoff/pull/181/commits/3de5e85d012f05f08cc66358a677be5d722b52d0